common/oatbox/log/logger/processor/UserIdProcessor.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\oatbox\log\logger\processor;
+
+use Psr\Log\LogLevel;
+
+/**
+ * Class UserIdProcessor
+ *
+ * @package oat\oatbox\log\logger\processor
+ */
+class UserIdProcessor
+{
+    /**
+     * @var string
+     */
+    protected $level;
+
+    /**
+     * EnvironmentProcessor constructor.
+     *
+     * @param string $level
+     */
+    public function __construct($level = LogLevel::DEBUG)
```

Comment:
If no options passed to constructor `$level` will be string. And it will cause int-string comparison. Logger::DEBUG contains correct integer representation.